Room/app/src/main/java/com/example/room/di/RepositoryModule.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.room.di
+
+import com.example.room.data.local.FriendRepositoryImpl
+import com.example.room.data.local.datasource.LocalFriendDataSource
+import com.example.room.domain.repository.FriendRepository
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ViewModelComponent
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object RepositoryModule {
+    @Provides
+    @Singleton
+    fun provideFriendRepository(friendDataSource: LocalFriendDataSource): FriendRepository =
+        FriendRepositoryImpl(friendDataSource)
+}
```

Comment:
따로 구현이 필요 없을 경우 **@Provide**대신 **abstract class**와 **@Binds**을 사용해서 간결하게 표현이 가능합니다~

```suggestion
abstract class RepositoryModule {
    @Binds
    abstract fun bindFriendRepository(friendRepositoryImpl: FriendRepositoryImpl): FriendRepository
}
```

Reply:
훨씬 간결하네요 감사합니다!!